### PR TITLE
Fixed math divide ceiling

### PR DESCRIPTION
**Pull request by cschuchardt88**
# Description

- Renamed method to `DivideCeiling`
- Fixed comments
- Optimized the method
- Now work with any number

Fixes # (issue)

## Type of change

- [x] Optimization (the change is only an optimization)
- [ ] Style (the change is only a code style for better maintenance or standard purpose)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
- [x] Unit Tests

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Neo.Extensions/BigIntegerExtensions.cs [hunk 1]
```diff
@@ -22,22 +22,20 @@ public static class BigIntegerExtensions
         /// Performs integer division with ceiling (rounding up).
         /// Example: 10 / 3 = 4 instead of 3.
         /// </summary>
-        /// <param name="a">The dividend.</param>
-        /// <param name="b">The divisor (must be greater than zero).</param>
+        /// <param name="dividend">The dividend.</param>
+        /// <param name="divisor">The divisor (must be greater than zero).</param>
```
Comment (ajara87) [inline src/Neo.Extensions/BigIntegerExtensions.cs]:
```suggestion
        /// <param name="divisor">The divisor (must be nonzero).</param>
```

File: src/Neo.Extensions/BigIntegerExtensions.cs [hunk 2]
```diff
@@ -22,22 +22,20 @@ public static class BigIntegerExtensions
...
         /// <returns>The result of division rounded up.</returns>
         /// <exception cref="ArgumentException">Thrown when divisor is zero or negative.</exception>
-        public static BigInteger CeilingDivide(this BigInteger a, BigInteger b)
+        public static BigInteger DivideCeiling(this BigInteger dividend, BigInteger divisor)
         {
```
Comment (ajara87) [inline src/Neo.Extensions/BigIntegerExtensions.cs:30]:
```suggestion
        if (divisor == 0)
            throw new ArgumentException("Divisor must be nonzero.", nameof(divisor));
```

File: src/Neo.Extensions/BigIntegerExtensions.cs [hunk 3]
```diff
@@ -22,22 +22,20 @@ public static class BigIntegerExtensions
         /// Performs integer division with ceiling (rounding up).
         /// Example: 10 / 3 = 4 instead of 3.
         /// </summary>
-        /// <param name="a">The dividend.</param>
-        /// <param name="b">The divisor (must be greater than zero).</param>
+        /// <param name="dividend">The dividend.</param>
+        /// <param name="divisor">The divisor (must be greater than zero).</param>
         /// <returns>The result of division rounded up.</returns>
         /// <exception cref="ArgumentException">Thrown when divisor is zero or negative.</exception>
```
Comment (ajara87) [inline src/Neo.Extensions/BigIntegerExtensions.cs]:
```suggestion
        /// <exception cref="ArgumentException">Thrown when divisor is zero.</exception>
```

**Review by ajara87 — CHANGES_REQUESTED** [review]

File: src/Neo.Extensions/BigIntegerExtensions.cs (hunk 2 above)
Comment (cschuchardt88) [reply src/Neo.Extensions/BigIntegerExtensions.cs:30]: It will automatically throw `DivideByZeroException` from `BigInteger.DivRem`

**Review by ajara87 — DISMISSED** [review]

File: tests/Neo.Extensions.Tests/UT_BigIntegerExtensions.cs [hunk 4]
```diff
@@ -24,52 +24,54 @@ public class UT_BigIntegerExtensions
...
         public void CeilingDivide_NegativeNumerator()
         {
-            var numerator = new BigInteger(-7);
-            var denominator = new BigInteger(3);
-            var result = BigIntegerExtensions.CeilingDivide(numerator, denominator);
-            Assert.AreEqual(-2, result);
+            var actual = BigIntegerExtensions.DivideCeiling(-7, 3);
+            Assert.AreEqual(-2, actual);
+
+            actual = BigIntegerExtensions.DivideCeiling(-7, -3);
```
Comment (shargon) [inline tests/Neo.Extensions.Tests/UT_BigIntegerExtensions.cs:30]: Please, add test with positive with negative values

Comment (cschuchardt88) [reply tests/Neo.Extensions.Tests/UT_BigIntegerExtensions.cs:30]: Request finished.

**Review by ajara87 — APPROVED** [review]

**Review by shargon — APPROVED** [review]

File: src/Neo.Extensions/BigIntegerExtensions.cs [hunk 5]
```diff
@@ -22,22 +22,20 @@ public static class BigIntegerExtensions
...
+        /// <param name="divisor">The divisor (must be nonzero).</param>
         /// <returns>The result of division rounded up.</returns>
-        /// <exception cref="ArgumentException">Thrown when divisor is zero or negative.</exception>
-        public static BigInteger CeilingDivide(this BigInteger a, BigInteger b)
+        /// <exception cref="ArgumentException">Thrown when divisor is zero.</exception>
+        public static BigInteger DivideCeiling(this BigInteger dividend, BigInteger divisor)
         {
-            if (b <= 0)
-                throw new ArgumentException("Divider must be greater than zero.", nameof(b));
+            var v = divisor > 0 ?
```
Comment (shargon) [inline src/Neo.Extensions/BigIntegerExtensions.cs:32]:
```suggestion
            // If it's 0, it will automatically throw DivideByZeroException
            var v = divisor > 0 ?
```
